xcape/core.py

```diff
@@ -129,40 +129,78 @@ def calc_cape(p, t, td, ps, ts, tds, **kwargs):
-        return _calc_cape_numpy(p, t, td, ps, ts, tds, **kwargs)
+        return _calc_cape_numpy(p, t, td, ps, ts, tds, *args, **kwargs)
     
 
-def _calc_cape_gufunc(p, t, td, ps, ts, tds, **kwargs):
-    signature = "(i),(i),(i),(),(),()->(),()"
-    output_dtypes = ('f4','f4')
-    if kwargs['source']=='most-unstable':
-        signature += ",(),()"
-        output_dtypes = output_dtypes + ('i4','f4')
-        
-    return da.apply_gufunc(_calc_cape_numpy, signature,
-                           p, t, td, ps, ts, tds,
-                           output_dtypes=('f4','f4'),
-                           axis=-1,
-                           vectorize=False,
-                           **kwargs)
-
+def _calc_cape_gufunc(p, t, td, ps, ts, tds, *args, **kwargs):
+    
+    if (kwargs['vertical_lev']=='sigma'):
+        signature = "(i),(i),(i),(),(),()->(),()"
+        output_dtypes = ('f4','f4')
+        if kwargs['source']=='most-unstable':
+            signature += ",(),()"
+            output_dtypes = output_dtypes + ('i4','f4')
+
+        return da.apply_gufunc(_calc_cape_numpy, signature,
+                               p, t, td, ps, ts, tds, 
+                               output_dtypes=('f4','f4'),
+                               axis=-1,
+                               vectorize=False,
+                               **kwargs)
+    
+    elif (kwargs['vertical_lev']=='pressure'):
+        print('here')
+        pres_lev_pos = args[0]
+        signature = "(i),(i),(i),(),(),(),()->(),()"
+        output_dtypes = ('f4','f4')
+        if kwargs['source']=='most-unstable':
+            signature += ",(),()"
+            output_dtypes = output_dtypes + ('i4','f4')
+
+        return da.apply_gufunc(_calc_cape_numpy, signature,
+                               p, t, td, ps, ts, tds, pres_lev_pos,
+                               output_dtypes=('f4','f4'),
+                               axis=-1,
+                               vectorize=False,
+                               **kwargs)
 
 # the numpy version of the algorithm
-def _calc_cape_numpy(p, t, td, ps, ts, tds, source='surface', ml_depth=500.,
-                     adiabat='pseudo-liquid',pinc=1000., method='fortran',
-                     vertical_lev='sigma', pres_lev_pos=1):
+def _calc_cape_numpy(p, t, td, ps, ts, tds, *args, 
+                     source='surface', ml_depth=500.,
+                     adiabat='pseudo-liquid',pinc=500., method='fortran',
+                     vertical_lev='sigma'):
 
 
     original_shape = p.shape
     original_surface_shape = ps.shape
 
     # after this, all arrays are 2d shape (nlevs, npoints)
     p_2d, t_2d, td_2d = _reshape_inputs(p, t, td)
-    p_s1d, t_s1d, td_s1d = _reshape_surface_inputs(ps, ts, tds)
+    
+    if len(args)==0:
+        pres_lev_pos=1
+        p_s1d, t_s1d, td_s1d = _reshape_surface_inputs(ps, ts, tds)
+    elif len(args)==1:
+        p_s1d, t_s1d, td_s1d, pres_lev_pos = _reshape_surface_inputs(ps, ts, tds, args[0])
 
```

Comment:
this is necessary, to add the reshaping of pres_lev_pos

Reply:
In python 3.6 it's not necessary. You an use variable args in assignment:
```
a, b, *c = 1, 2, 3, 4, 5
```

Moreover, it seems like the whole thing would be simpler if all variables were in `args`.